queries/sun_position.py

```diff
@@ -0,0 +1,623 @@
+    result["date"] = datetime.date.today() + datetime.timedelta(days=1)
+
+
+### QSunLocation ###
+
+
+def QSunCapitalRegion(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["city"] = "Reykjavík"
+
+
+def QSunArbitraryLocation(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["city"] = capitalize_placename(result._nominative)
+
+
+###
+
+_ALMANAK_HI_URL: str = "http://www.almanak.hi.is/solgang.html"
+_ALMANAK_HI_COLUMNS: Tuple[_SOLAR_POS_ENUM, ...] = (
+    _SOLAR_POSITIONS.DÖGUN,
+    _SOLAR_POSITIONS.BIRTING,
+    _SOLAR_POSITIONS.SÓLRIS,
+    _SOLAR_POSITIONS.HÁDEGI,
+    _SOLAR_POSITIONS.SÓLARLAG,
+    _SOLAR_POSITIONS.MYRKUR,
+    _SOLAR_POSITIONS.DAGSETUR,
+    _SOLAR_POSITIONS.SÓLARHÆÐ,
+)
+
+
+# Match lines such as (lat/lon given in format DD°MM.M')
+#    "VESTMANNAEYJAR  2021   63°26,2'  20°16,5'"
+_ALMANAK_CITY_REGEX = re.compile(
+    r"^(\w+)\s+\d\d\d\d\s+(\d+°\d+,\d+')\s+(\d+°\d+,\d+'?)$"
+)
+
+# Match lines such as:
+#    "JAN  1.    07 55    10 03    11 18    13 31    15 45    17 00    19 08     3,1"
+#    " -   8.    07 52    09 57    11 08    13 35    16 01    17 13    19 18     3,9"
+#    " -  29.                      09 02    13 11    16 00                      12,3"
+
+_ALMANAK_SUNPOS_REGEX = re.compile(
+    r"^(\w\w\w| - )\s+(\d+)\." + 7 * r"    (\d\d \d\d|     )" + r"\s+(\d+,\d+)$"
+)
+
+
+def _convert_dms_lat_lon_to_decimal(lat: str, lon: str) -> LatLonTuple:
+    """
+    Convert lat/lon coordinate strings of the format DD°MM.M' to decimal degrees.
+    Returns tuple of floats.
+    """
+    dlat_str, mlat_str = lat.rstrip("'").replace(",", ".").split("°")
+    dlat: float = float(dlat_str) + (float(mlat_str) / 60)
+
+    dlon_str, mlon_str = lon.rstrip("'").replace(",", ".").split("°")
+    dlon: float = float(dlon_str) + (float(mlon_str) / 60)
+
+    return (dlat, dlon)
+
+
+_SOLAR_CELL_TYPE = Union[datetime.time, float, None]
+
+
+def _parse_almanak_cell(pt_str: str) -> _SOLAR_CELL_TYPE:
+    """
+    Parse a cell in Almanak HÍ. Returns datetime.time for cells containing
+    a timestamp, float for cells containing the solar height and None otherwise.
+    Examples:
+        '03 15' => datetime.time(3,15)
+        '34,9'  => 34.9
+        '     ' => None
+    """
+    if not pt_str.isspace():
+        if " " in pt_str:
+            # Column contains a time value, convert to datetime
+            hour, minute = pt_str.split()
+            return datetime.time(hour=int(hour) % 24, minute=int(minute) % 60)
+
+        if "," in pt_str:
+            # Column contains solar height at solar noon (in degrees)
+            return float(pt_str.replace(",", "."))
+
+    return None
+
+
+_SOLAR_ROW_TYPE = Dict[_SOLAR_POS_ENUM, _SOLAR_CELL_TYPE]
+_SOLAR_DICT_TYPE = Dict[
+    str, Dict[Union[str, datetime.date], Union[LatLonTuple, _SOLAR_ROW_TYPE]]
+]
+
+
+def _parse_almanak_hi_data(text: List[str]) -> _SOLAR_DICT_TYPE:
```

Comment:
Strictly speaking the type can be Iterable[str] (which is less strict and allows more inputs)

Reply:
Fixed!